pkg/chunk/aws/sse_config.go

```diff
@@ -0,0 +1,68 @@
+package aws
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// SSEKMS config type constant to configure S3 server side encryption using KMS
+	// https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html
+	SSEKMS     = "SSE-KMS"
+	sseKMSType = "aws:kms"
+	// SSES3 config type constant to configure S3 server side encryption with AES-256
+	// https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html
+	SSES3     = "SSE-S3"
+	sseS3Type = "AES256"
+)
+
+// SSEEncryptionConfig configures server side encryption (SSE)
+type SSEEncryptionConfig struct {
+	ServerSideEncryption string
+	KMSKeyID             *string
+	KMSEncryptionContext *string
```

Comment:
I'm not a big fan of pointers in config structs. You can achieve the same assuming empty string means not configured (which should be fine in this context).

Reply:
AWS lib uses string pointers several times in there. Removing string pointers from `sse_config.go (SSEEncryptionConfig)` means that I will have to treat that on the file `s3_storage_client.go`, which has more logic that should have.